dev_guide/application_lifecycle.adoc

```diff
@@ -0,0 +1,93 @@
+= The Application Life Cycle
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+As a PaaS, OpenShift is built for building and deploying applications. Depending
+on how much you want to involve OpenShift in the development process, you can
+choose to develop on OpenShift after it has been installed, and use it to
```

Comment:
@bfallonf Assuming "after it has been installed" refers to OpenShift, I think we can drop that bit. Seems redundant for these Developer docs where the overall assumption is they have an OpenShift instance to access.